Board.cpp

```diff
@@ -288,6 +334,23 @@ set<int> Board::genMoves() const
 	return moves;
 }
 
+set<int> Board::generatePawnPromotionMoves(uint8_t from, uint8_t to) 
+{
+
+	set<int> promotionMoves;
+	Move promotionMove;
+	
+	//1,2,4,8 represent N,B,R,Q
+	for(uint8_t i = 1; i <=8; i*=2)
```

Comment:
Just a tip. You can do left shift by 1  i.e. ( i << 1) instead of (i *=2) . Its nbd but it does save a few clock cycles and is the idiomatic way to do a for loop  over bits :)